### DISCUSSION TITLE
Get entry files in Transformer plugin

### DISCUSSION THREAD

**Discussion by hellos3b** [parcel-bundler/parcel#9245] [Q&A] [answered]
I'm trying to create a plugin similar to html where additional dependencies are added  based on the contents of the entry files. But my plugin works off of `*.js` files, so I need a way to "skip" any files that weren't provided in the initial list

Is there a way to get a list of the files that were specified to parcel, either through `"sources"` in package.json or through cli `npx parcel main.js`?

**Comment by mischnic** [comment, marked as answer]
That is currently not possible in a transformer, I guess because these entry files could also be imported by something else in your project, so the cases are actually "entry", "entry but also imported", and "imported".

Resolvers do have access to this information (`dependency.sourcePath == null` if it's an entry). The question is how you then forward this from the resolver to a transformer. Maybe with [`query`](https://github.com/parcel-bundler/parcel/blob/3d7a35369de462c0966b610f0c7f1d877490d912/packages/core/types/index.js#L1586) which is then available in the transformer as `asset.query`.

**Reply by hellos3b** [reply]
Thanks @mischnic ! Gave this a try, and after some trial and error the `query` worked as long as `fullPath` is also provided
